# Ejemplo 3.5.2 de Díaz y Morales (2015)

<img src="https://raw.githubusercontent.com/fhernanb/fhernanb.github.io/master/my_docs/logo_unal_color.png" alt="drawing" width="200"/>

En este ejemplo se desea estudiar el siguiente conjunto de hipótesis usando un nivel de significancia de 0.05.

$H_0: \boldsymbol{\mu}_1 = \boldsymbol{\mu}_2$

$H_A: \boldsymbol{\mu}_1 \neq \boldsymbol{\mu}_2$

Asumiendo que las matrices $\boldsymbol{\Sigma}_1$ y $\boldsymbol{\Sigma}_2$ son desconocidas pero que se pueden suponer iguales.

Las matrices muestrales son

\begin{equation}
\bar{\mathbf{X}}_1 = 
\begin{pmatrix}
15.97 \\
15.91 \\
27.19 \\
22.75
\end{pmatrix}
\end{equation}

\begin{equation}
\bar{\mathbf{X}}_2= 
\begin{pmatrix}
12.34 \\
13.91 \\
16.59 \\
21.94
\end{pmatrix}
\end{equation}

\begin{equation}
\mathbf{S}_1 = 
\begin{pmatrix}
5.192 & 4.545 & 6.522 & 5.250 \\
4.545 & 13.184 & 6.760 & 6.266 \\
6.522 & 6.760 & 28.67 & 14.468 \\
5.250 & 6.266 & 14.468 & 16.645
\end{pmatrix}
\end{equation}

\begin{equation}
\mathbf{S}_2 = 
\begin{pmatrix}
9.136 & 7.549 & 5.531 & 4.151 \\
7.549 & 18.60 & 5.446 & 5.446 \\
5.531 & 5.446 & 13.55 & 13.55 \\
4.151 & 5.446 & 13.55 & 28
\end{pmatrix}
\end{equation}

con $n_1=32$ y $n_2=32$ sujetos por cada grupo.

<img src="ph_dos_mu_con_sigma_desco_pero_iguales.png" alt="" width="600"/>

In [1]:
import numpy as np
import scipy.stats

A continuación los datos del ejemplo

In [2]:
xbarra_1 = np.matrix([[15.97],
                      [15.91],
                      [27.19],
                      [22.75]])

xbarra_2 = np.matrix([[12.34],
                      [13.91],
                      [16.59],
                      [21.94]])

S1 = np.matrix([[5.192, 4.545, 6.522, 5.250],
                [4.545, 13.184, 6.760, 6.266],
                [6.522, 6.760, 28.67, 14.468],
                [5.250, 6.266, 14.468, 16.645]])

S2 = np.matrix([[9.136, 7.549, 5.531, 4.151],
                [7.549, 18.60, 5.446, 5.446],
                [5.531, 5.446, 13.55, 13.55],
                [4.151, 5.446, 13.55, 28]])

n1 = 32
n2 = 32

p = 4   # número de variables

Primero vamos a calcular la matriz combinada $\mathbf{S}_p$.

In [3]:
Sp = (n1-1) * S1 / (n1+ n2 - 2) + (n2-1) * S2 / (n1+ n2 - 2)
print(Sp)

[[ 7.164   6.047   6.0265  4.7005]
 [ 6.047  15.892   6.103   5.856 ]
 [ 6.0265  6.103  21.11   14.009 ]
 [ 4.7005  5.856  14.009  22.3225]]


Ahora calcularmos $T^2$.

In [4]:
aux = xbarra_1 - xbarra_2
T2 = np.matmul(np.matmul(aux.T, np.linalg.inv(Sp)), aux)
T2 = T2 * (n1*n2) / (n1+n2)
T2.item()  # Para obtener el T2 como un escalar y no como una matriz

136.6716140046685

Para calcular el valor crítico con un $\alpha=0.05$ hacemos lo siguiente

In [5]:
scipy.stats.f.ppf(q=1-0.05, dfn=p, dfd=n1+n2-p-1) * (n1+n2-2) * p / (n1+n2-p-1)  # Valor crítico

10.625776750234012

<div class="alert alert-success">
  <strong>Resultado</strong> Como el estadístico es $T^2=136.67$, que es mayor que el valor crítico $10.63$, entonces hay evidencias suficientes para rechazar $H_0: \boldsymbol{\mu}_1 = \boldsymbol{\mu}_2$.
</div>